app/client/graphQL/entry.ts

```diff
@@ -0,0 +1,338 @@
+        return response;
+    }
+
+    let members = response.data.channelMembers;
+
+    while (members?.length === MEMBERS_PER_PAGE) {
+        let pageResponse;
+        try {
+            // eslint-disable-next-line no-await-in-loop
+            pageResponse = await gqlAllChannelsNextPage(serverUrl, members[members.length - 1].cursor!);
+        } catch {
+            break;
+        }
+
+        if ('error' in pageResponse || 'errors' in pageResponse) {
+            break;
+        }
+
+        members = pageResponse.data.channelMembers!;
+        response.data.channelMembers?.push(...members);
+    }
+
+    return response;
+};
+
+const gqlAllChannelsNextPage = async (serverUrl: string, cursor: string) => {
+    return doGQLQuery(serverUrl, nextPageAllChannelsQuery(cursor), 'channelsNextPage');
+};
+
+const entryQuery = `
+query entry {
+    config
+    license
+    user(id:"me") {
+        id
+        authService
+        deleteAt
+        email
+        #updateAt
+        firstName
+        lastName
+        lastPictureUpdateAt
+        locale
+        nickname
+        position
+        roles {
+            id
+            name
+            permissions
+        }
+        username
+        notifyProps
+        props
+        timezone
+        isBot
+        status {
+            status
+        }
+        preferences{
+            category
+            name
+            value
+            userId
+        }
+        sessions {
+            createAt
+            expiresAt
+        }
+    }
+    teamMembers(userId:"me") {
+        deleteAt
+        roles {
+            id
+            name
+            permissions
+        }
+        team {
+            id
+            description
+            displayName
+            name
+            type
+            allowedDomains
+            lastTeamIconUpdate
+            groupConstrained
+            allowOpenInvite
+            updateAt
+        }
+    }
+}
+`;
+
+const channelsQuery = (teamId: string, exclude: boolean) => {
+    return `
+query channels {
+    channelMembers(userId:"me", first:PER_PAGE, teamId:"TEAM_ID", excludeTeam:EXCLUDE) {
+        cursor
+        msgCount
+        mentionCount
+        lastViewedAt
+        notifyProps
+        roles {
+            id
+            name
+            permissions
+        }
+        channel {
+            id
+            header
+            purpose
+            type
+            createAt
+            creatorId
+            deleteAt
+            displayName
+            prettyDisplayName
+            groupConstrained
+            name
+            shared
+            lastPostAt
+            totalMsgCount
+            team {
+                id
+            }
+        }
+    }
+    sidebarCategories(userId:"me", teamId:"TEAM_ID", excludeTeam:EXCLUDE) {
+        displayName
+        id
+        sorting
+        type
+        muted
+        collapsed
+        channelIds
+        teamId
+    }
+}
+`.replace('PER_PAGE', `${MEMBERS_PER_PAGE}`).
```

Comment:
As I also mentioned in our meeting, this seems a bit dubious, you can just put `${MEMBERS_PER_PAGE}` inline in the string template literal. Intuitively I'm still a bit scared of the GraphQL equivalent of SQL injections with this style, but as you pointed out all values passed here are internally generated/verified.

Reply:
We discussed this. Daniel will be using variable binding: https://graphql.org/learn/queries/#variables to fix this properly.